File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/WordDomainMapper.java
package com.przemyslawlusnia.vocabularycreator.wordlist.domain;

import com.przemyslawlusnia.vocabularycreator.wordlist.repository.WordRealm;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class WordDomainMapper {

  public WordDomainModel mapToWordDomainModel(WordRealm wordRealm) {
    return new WordDomainModel(wordRealm.getWord(), wordRealm.getTranslation(), wordRealm.getType());
  }

  public List<WordDomainModel> mapToWordDomainModels(List<WordRealm> wordsRealm) {
    if (wordsRealm == null) {
      return Collections.emptyList();
    }
    List<WordDomainModel> result = new ArrayList<>(wordsRealm.size());
    for (WordRealm wordRealm : wordsRealm) {
      result.add(new WordDomainModel(wordRealm.getWord(), wordRealm.getTranslation(), wordRealm.getType()));
    }
    return result;
  }

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/WordDomainMapper.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.domain

import com.przemyslawlusnia.vocabularycreator.wordlist.repository.WordRealm

class WordDomainMapper {

    fun mapToWordDomainModel(wordRealm: WordRealm): WordDomainModel {
        return WordDomainModel(wordRealm.word, wordRealm.translation, wordRealm.type)
    }

    fun mapToWordDomainModels(wordsRealm: List<WordRealm>): List<WordDomainModel> {
        return wordsRealm.map { WordDomainModel(it.word, it.translation, it.type) }
    }

}
